taskvine/src/manager/vine_manager.c

```diff
@@ -4586,30 +4555,14 @@ static struct vine_task *send_library_to_worker(struct vine_manager *q, struct v
 	vine_result_code_t result = commit_task_to_worker(q, w, t);
 
 	/* Careful: If this failed, then the worker object may longer be valid! */
-
 	if (result == VINE_SUCCESS) {
+		vine_txn_log_write_library_update(q, w, t->task_id, VINE_LIBRARY_SENT);
 		return t;
 	} else {
 		return 0;
```

Comment:
What happens on failure here?
Is it necessary to delete the task and back it out of the table, or does that happen naturally as part of the task state evolution from commit_task_to_worker?
(Either way some commentary may help here.)